frontend/src/main/java/com/google/tripmeout/frontend/servlets/ServletUtil.java

```diff
@@ -17,37 +15,25 @@
  * Static utilities that provide functionality that may be shared between servlets.
  */
 public final class ServletUtil {
-  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
 
-  public static <T> Optional<T> extractFromRequestBody(
-      HttpServletRequest request, Gson gson, Class<T> clazz) throws IOException {
-    try (Reader reader = request.getReader()) {
-      return Optional.ofNullable(gson.fromJson(reader, clazz));
-    } catch (JsonParseException e) {
-      logger.atWarning().withCause(e).log(
-          "received exception while attempting to parse json from reader!");
-      return Optional.empty();
-    }
+  public static <T> Optional<T> extractFromRequestBody(HttpServletRequest request, Gson gson,
```

Comment:
We're letting exceptions from `gson.fromJson()` escape here, so the optional should never be empty.

If the overall request type is `null`, then we likely want to throw an exception here anyway (instead of making the caller check/throw in-line at every site). Consider adding a new exception type  to indicate a bad request (or co-opting `WrongFormatUriException` for a more general `BadRequestException`, depending on the granularity of error handing/reporting we expect to have).

Reply:
I'm a little confused here: if I catch JsonParseExceptions and set the status to 400, do I still need a separate checked exception? and shouldn't a null request lead to a JsonParseException?